src/antenna.c

```diff
@@ -75,21 +104,50 @@ void deploy_antenna(void) {
     }
 
     // Start Algorithm 2 for not opened doors
-    write_antenna_alg2(doors_to_redeploy, &i2c_status);
+    uint8_t ret_status = write_antenna_alg2(doors_to_redeploy, &i2c_status);
     read_antenna_data(door_positions, &mode, main_heaters, backup_heaters, &timer_s, &i2c_status);
+    // Wait ~35seconds per door for algorithm to finish
     timer = 35 * num_doors;
 #ifdef ANTENNA_DEBUG
     print_bytes(&doors_to_redeploy, 1);
 #endif
     while (mode != 0 && timer > 0) {
         WDT_ENABLE_SYS_RESET(WDTO_8S);
-        _delay_ms(1000); //Wait ~35seconds for algorithm to finish
+        _delay_ms(1000);
         timer -= 1;
         read_antenna_data(door_positions, &mode, main_heaters, backup_heaters, &timer_s, &i2c_status);
     }
 
+    // Check which doors are still open
+    read_antenna_data(door_positions, &mode, main_heaters, backup_heaters, &timer_s, &i2c_status);
```

Comment:
I think you need to save the return value into `ret_status` here?

Reply:
Sure that works too. I only saved the return status for a write command because I didn't think it mattered if the read failed, but I can do it for the read instead.